nixos/modules/services/taler/exchange.nix

```diff
@@ -0,0 +1,184 @@
+{
+  lib,
+  config,
+  pkgs,
+  ...
+}:
+
+let
+  this = config.services.taler.exchange;
+  # Services that need access to the DB
+  servicesDB = [
+    "httpd"
+    "aggregator"
+    "closer"
+    "wirewatch"
+  ];
+  # Services that do not need access to the DB
+  servicesNoDB = [
+    "secmod-cs"
+    "secmod-eddsa"
+    "secmod-rsa"
+  ];
+  services = servicesDB ++ servicesNoDB;
+  dbName = "taler-exchange-httpd";
+  # taler-exchange needs a runtime dir shared between the taler services. Crypto
+  # helpers put their sockets here for instance and the httpd connects to them.
+  runtimeDir = "/run/taler-system-runtime/";
+  talerEnabled = config.services.taler.enable;
+  inherit (config.services.taler) configFile;
+in
+
+{
+  options.services.taler.exchange = {
+    enable = lib.mkEnableOption "the GNU Taler exchange";
+    package = lib.mkPackageOption pkgs "taler-exchange" { };
+    denominationConfig = lib.mkOption {
+      type = lib.types.lines;
+      default = throw "You must set the denomination config `services.taler.exchange.denominationConfig`.";
```

Comment:
We cannot use throw here

Reply:
Why not? This is an important option which the program does not correctly work without.

The best way to mandate that the user sets it is this way.

Reply:
It would abort the evaluation and it is best to use assertions/warnings to get them all at once.

Reply:
> it is best to use assertions/warnings to get them all at once

Is there an example of how I can do this?